client/src/components/advancedSearch/Condition.tsx

```diff
@@ -0,0 +1,271 @@
+            : regInt.test(conditionValue)
+        );
+        break;
+      case 'float':
+        setIsValueLegal(
+          isIn
+            ? regFloatInterval.test(conditionValue)
+            : regFloat.test(conditionValue)
+        );
+        break;
+      default:
+        setIsValueLegal(false);
+        break;
+    }
+
+    triggerChange({
+      field: conditionField,
+      op: operator,
+      value: conditionValue,
+      isCorrect: isValuelegal,
+      id,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [conditionField, operator, conditionValue]);
+
+  // Trigger change event if isValuelegal changed.
+  useEffect(() => {
+    triggerChange({
+      field: conditionField,
+      op: operator,
+      value: conditionValue,
+      isCorrect: isValuelegal,
+      id,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isValuelegal]);
+
+  const classes = useStyles();
+
+  // Logic operator input change.
+  const handleOpChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setOperator(event.target.value);
+  };
+  // Field Name input change.
+  const handleFieldNameChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const value = event.target.value;
+    const target = fields.find(field => field.name === value);
+    target && setConditionField(target);
+  };
+  // Value input change.
+  const handleValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    setConditionValue(value);
+  };
+
+  return (
+    <div className={`${classes.wrapper} ${className}`} {...others}>
+      <TextField
+        className={classes.fieldName}
+        label="Field Name"
+        variant="filled"
+        size="small"
+        select
+        onChange={handleFieldNameChange}
+        value={conditionField?.name}
+      >
+        {fields.map((field: Field) => (
+          <MenuItem key={field.name} value={field.name}>
+            {field.name}
+          </MenuItem>
+        ))}
+      </TextField>
+      <TextField
+        className={classes.logic}
+        label="Logic"
+        variant="filled"
+        size="small"
+        select
+        onChange={handleOpChange}
+        defaultValue={LogicalOperators[0].value}
+        value={operator}
+      >
+        {LogicalOperators.map(option => (
+          <MenuItem key={option.value} value={option.value}>
+            {option.label}
+          </MenuItem>
+        ))}
+      </TextField>
+      <TextField
+        className={classes.value}
+        label="Value"
+        variant="filled"
+        size="small"
+        onChange={handleValueChange}
+        value={conditionValue}
+        error={!isValuelegal}
+      />
+      <IconButton
+        aria-label="close"
+        className={classes.closeButton}
+        onClick={onDelete}
+        size="small"
+      >
+        <CloseIcon />
+      </IconButton>
+    </div>
+  );
+};
+
+Condition.propTypes = {
+  /**
```

Comment:
we have typescript, so do we need propTypes here?

Reply:
Ts validates types at compile time, whereas PropTypes are checked at runtime.
Here remove PropTypes.